app/controllers/api/admin_controller.rb

```diff
@@ -0,0 +1,85 @@
+class Api::AdminController < Api::ApplicationController
+  before_action :admin_user
+
+  def view_profile
+    @user = User.find(params[:id])
+
+    render json: @user, serializer: UserSerializer
+  end
+
+  def view_users
+    @users = User.all
+
+    render json: @users, each_serializer: UserSerializer
+  end
+
+  def destroy_user
+    User.find(params[:id]).destroy
+
+    render json: { success: 'User deleted' }, status: :ok
+  end
+
+  def view_products
+    @products = Product.includes(:category)
+
+    render json: @products, each_serializer: ProductSerializer
+  end
+
+  def create_product
+  end
+
+  def edit_product
+    @product = Product.find(params[:id])
+
+    render json: @product, serializer: ProductSerializer
+  end
+
+  def update_product
+    @product = Product.find(params[:id])
```

Comment:
You have multiple places where you are doing this, so you could extract it to a method, `set_product`, & do `before_action :set_product, only: %i[edit_product update_product etc..]`